### PR TITLE
Greeter bot for new contributors

### LINKED ISSUE #1007: Implement bot to provide helpful comment on first contributions
An idea that was brought up at out last community meeting is to create a bot (presumably through GitHub Actions) that would provide a helpful welcome message as a comment the first time that someone makes a pull request.  

The welcome message should link to our development guide and provide information on or links about:
 - Writing documentation (with links to guides on Sphinx and reStructuredText)
 - Writing tests (with links to pytest, plus our test helpers when they're implemented)
 - pre-commit
 - A guide on git and GitHub
 - The code review process
 - Our communication channels and meetings
 - Our code of conduct

We do want this to provide a warm welcome and an expression of gratitude, of course.  We should also try to be succinct and not provide unnecessary detail.  Thanks!

### PR DESCRIPTION
**Pull request by StanczakDominik**
Closes #1007.

I guess the only unexpected thing in this PR is how to write multiline strings in yaml files.

Compared to #1007, this does not explicitly say much about how to actually do stuff, because I figure the contributor guide (which is overdue for improvements, anyway) should be the central point for that and I'd rather not duplicate those here.

I'm open to adding a few more links e.g. to particular parts of the dev guide if that's deemed helpful, though.

### CODE REVIEW

File: .github/workflows/greetings.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          Otherwise, we've got [a development
+          guide](https://docs.plasmapy.org/en/latest/development/index.html)
+          you might find helpful.
+
+          * We try to be the best community we possibly can be. As part of that
+          effort, we have a [Code of
+          Conduct](https://docs.plasmapy.org/en/latest/CODE_OF_CONDUCT.html)
+          that all contributors should follow.
+
+          And once more, thanks a ton and welcome to the community!
```
Comment (namurphy) [inline .github/workflows/greetings.yml:41]:
```suggestion
          And once more, thanks a 907.2 kg at standard Earth gravity and welcome to the community!
```
SI units!  😹  😹  😬

File: .github/workflows/greetings.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+          * the "Checks" tab at the bottom of the pull request page runs our
+          test suite. Don't worry if something broke! We break stuff all the
+          time. Eventually, we'll want to get tests on this PR to pass before
+          merging it, though, so if you want to play detective (it's fun, try
+          it), you can start looking through any failing tests' logs and seeing
+          why they did. Of course, feel free to ask for help if you're having
+          trouble (we do that all the time, as well).
+
+          * if you'd like some real time help, the easiest way to do so is to
```
Comment (namurphy) [inline .github/workflows/greetings.yml:29]:
```suggestion
          * The "Checks" tab at the bottom of the pull request page runs our
          test suite. Don't worry if something broke! We break stuff all the
          time. Eventually, we'll want to get tests on this PR to pass before
          merging it, though, so if you want to play detective (it's fun, try
          it), you can start looking through any failing tests' logs and seeing
          why they did. Of course, feel free to ask for help if you're having
          trouble (we do that all the time, as well).

          * If you'd like some real time help, the easiest way to do so is to
```
for consistency with the third bullet

File: .github/workflows/greetings.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        # issue-message: 'not currently posting messages for first interaction via issue'
+        pr-message: >+
+          Thanks for opening your first PR to PlasmaPy!
+
+          Someone should get around to looking into this PR soon. In the meantime,
+          there are a few things you could do:
+
+          * the "Checks" tab at the bottom of the pull request page runs our
+          test suite. Don't worry if something broke! We break stuff all the
+          time. Eventually, we'll want to get tests on this PR to pass before
```
Comment (namurphy) [inline .github/workflows/greetings.yml:23]: This is definitely a good message to convey.  Things probably won't work the first time, but that's totally fine.

**Review by namurphy — CHANGES_REQUESTED** [review]
Thank you for this one too!  I was also thinking it'd be helpful to discuss the code review process, in particular since that can sometimes be unexpected and intimidating for new contributors.

File: .github/workflows/greetings.yml (hunk 1 above)
Comment (StanczakDominik) [reply .github/workflows/greetings.yml:41]: Huh. You know, I always thought that was a unit of mass, not of... weight...?

Comment (StanczakDominik) [reply .github/workflows/greetings.yml:41]: Can we do megagram, instead?

File: .github/workflows/greetings.yml (hunk 3 above)
Comment (StanczakDominik) [reply .github/workflows/greetings.yml:23]: Yeah; I thought we should have that in here based on Christoph's adventure with the PR template. That one needs changing, too...

**Comment by StanczakDominik** [comment]
I added a brief note on code review, but I'm not entirely sure what you'd like this to say.

**Comment by namurphy** [comment]
I just remembered this and am wondering if we could get this merged fairly soon, since I really like this idea.  Would it be okay if I worked on this a bit?  I was thinking that it would be good to add links to the contributor guide (especially the doc guide, test guide, and soon-to-be-updated changelog guide).

**Comment by StanczakDominik** [comment]
Absolutely, if you'd like to pick it up, feel free to! You're probably the best person for the job anyway, as you've got the best idea of where to point people to 😅

**Comment by StanczakDominik** [comment]
pre-commit.ci autofix

**Comment by StanczakDominik** [comment]
Ah, let's just try it out. :>

**Review by namurphy — APPROVED** [review]

**Pull request merged**